src/wallet/rpcdump.cpp

```diff
@@ -702,8 +702,36 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
     file << "\n";
 
     // add the base58check encoded extended master if the wallet uses HD
-    CKeyID masterKeyID = pwalletMain->GetHDChain().masterKeyID;
-    if (!masterKeyID.IsNull())
+    CHDChain chain = pwalletMain->GetHDChain();
+    CKeyID masterKeyID = chain.masterKeyID;
+    if(!chain.IsNull())
```

Comment:
@levonpetrosyan93 I think there should also be a condition check here for HDChain version >= `VERSION_WITH_BIP39`

Reply:
Good catch. Thank you.